src/spring-cloud/azext_spring_cloud/gateway.py

```diff
@@ -0,0 +1,225 @@
+
+    return _create_or_update_gateway_route_configs(client, resource_group, service, name, app_resource_id, routes)
+
+
+def gateway_route_config_update(cmd, client, resource_group, service, name,
+                                app_name=None,
+                                routes_json=None,
+                                routes_file=None):
+    gateway_route_config = client.gateway_route_configs.get(
+        resource_group, service, DEFAULT_NAME, name)
+
+    app_resource_id = _update_app_resource_id(client, resource_group, service, app_name, gateway_route_config.properties.app_resource_id)
+    routes = _update_routes(routes_file, routes_json, gateway_route_config.properties.routes)
+
+    return _create_or_update_gateway_route_configs(client, resource_group, service, name, app_resource_id, routes)
+
+
+def gateway_route_config_remove(cmd, client, resource_group, service, name):
+    return client.gateway_route_configs.begin_delete(resource_group, service, DEFAULT_NAME, name)
+
+
+def _update_api_metadata(existing, api_title, api_description, api_documentation_location, version, server_url):
+    if api_title is None and api_description is None and api_documentation_location is None and version is None and server_url is None:
+        return None
+    api_metadata = models.GatewayApiMetadataProperties() if existing is None else existing
+    if api_title:
+        api_metadata.title = api_title
+    if api_description:
+        api_metadata.description = api_description
+    if api_documentation_location:
+        api_metadata.documentation = api_documentation_location
+    if version:
+        api_metadata.version = version
+    if server_url:
+        api_metadata.server_url = server_url
+    return api_metadata
+
+
+def _update_cors(existing, allowed_origins, allowed_methods, allowed_headers, max_age, allow_credentials, exposed_headers):
+    if allowed_origins is None and allowed_methods is None and allowed_headers is None and max_age is None and allow_credentials is None and exposed_headers is None:
+        return None
+    cors = existing if existing is not None else models.GatewayCorsProperties()
+    if allowed_origins:
+        cors.allowed_origins = allowed_origins.split(",")
+    if allowed_methods:
+        cors.allowed_methods = allowed_methods.split(",")
+    if allowed_headers:
+        cors.allowed_headers = allowed_headers.split(",")
+    if max_age:
+        cors.max_age = max_age
+    if allow_credentials is not None:
+        cors.allow_credentials = allow_credentials
+    if exposed_headers:
+        cors.exposed_headers = exposed_headers.split(",")
+    return cors
+
+
+def _validate_route_config_exist(client, resource_group, service, name):
+    route_configs = client.gateway_route_configs.list(
+        resource_group, service, DEFAULT_NAME)
+    if name in (route_config.name for route_config in list(route_configs)):
+        raise CLIError("Route config " + name + " already exists")
```

Comment:
Could we use the specific error type `InvalidArgumentValueError` or `ValidationError` instead of `CLIError`?

Reply:
Updated and added description